formats/hocon/src/main/kotlin/kotlinx/serialization/hocon/HoconDecoder.kt

```diff
@@ -0,0 +1,49 @@
+package kotlinx.serialization.hocon
+
+import com.typesafe.config.Config
+import kotlinx.serialization.ExperimentalSerializationApi
+
+/**
+ * Decoder used by Hocon during deserialization.
+ * This interface allows to call methods from the Lightbend/config library on the [Config] object.
+ *
+ * Usage example (nested config serialization):
+ * ```
+ * @Serializable
+ * data class Example(
+ *   @Serializable(NestedConfigSerializer::class)
+ *   val d: Config
+ * )
+ *
+ * object NestedConfigSerializer : KSerializer<Config> {
+ *     override val descriptor: SerialDescriptor = PrimitiveSerialDescriptor("package.Config", PrimitiveKind.STRING)
+ *
+ *     override fun deserialize(decoder: Decoder): Config = if (decoder is HoconDecoder) {
+ *         decoder.decodeValue { conf, path -> conf.getConfig(path) // call to Lightbend/config library  }
+ *     } else throw SerializationException("This class can be decoded only by Hocon format")
+ *
+ *     override fun serialize(encoder: Encoder, value: Config) {
+ *         if (encoder is AbstractHoconEncoder) {
+ *             encoder.encodeConfigValue(value.root())
+ *         } else throw SerializationException("This class can be encoded only by Hocon format")
+ *     }
+ * }
+ *
+ * val nestedConfig = ConfigFactory.parseString("nested { value = \"test\" }")
+ * val globalConfig = Hocon.encodeToConfig(Example(nestedConfig)) // d: { nested: { value = "test" } }
+ * val newNestedConfig = Hocon.decodeFromConfig(Example.serializer(), globalConfig)
+ * ```
+ */
+@ExperimentalSerializationApi
+sealed interface HoconDecoder {
+
+    /**
+     * Decodes the next value in the current input.
+     * Allows calling methods from [Config].
+     *
+     * @param E type of serialize value
```

Comment:
```suggestion
     * @param E type of value
```

Not sure what you mean here under 'serialize value'

Reply:
Fixed